### ISSUE TITLE
[BUG]: `fatal error: 'fast_float/fast_float.h' file not found` During Compilation

### ISSUE THREAD

**Issue by ThisNekoGuy** [PCSX2/pcsx2#7093]
### Describe the Bug

Compiling `v1.7.3341-0-g622914830` with Clang throws an error early in compilation about `fast_float/fast_float.h` being missing:
```
. . .
pcsx2/common/../common/StringUtil.h:28:10: fatal error: 'fast_float/fast_float.h' file not found
#include "fast_float/fast_float.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
. . .
```

### Reproduction Steps

Steps I took, represented in bash:
```bash
  export CC=clang
  export CXX=clang++
  export LD="/usr/bin/lld-link"
  export AR="/usr/bin/llvm-ar"
  export NM="/usr/bin/llvm-nm"
  export AS="/usr/bin/llvm-as"
  export RANLIB="/usr/bin/llvm-ranlib"
  export STRIP="/usr/bin/llvm-strip"
  export OBJCOPY="/usr/bin/llvm-objcopy"
  export useClang=1
  export clangTidy=1

  export CFLAGS="-march=znver2 -mtune=znver2 -O3 -pipe -fno-plt -minline-all-stringops -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security \
        -fstack-clash-protection -fstack-protector-strong -fcf-protection"
  export CXXFLAGS="${CFLAGS} -Wp,-D_GLIBCXX_ASSERTIONS"
  export LDFLAGS="-unwind=libunwind -Wl,-O3,--sort-common,--as-needed,-z,relro,-z,now"

  mkdir -p build
  cd build

  cmake ../pcsx2 \
      -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DUSE_SYSTEM_YAML=TRUE \
      -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
      -DWAYLAND_API=ON \
      -USE_VULKAN=ON \
      -GNinja \
      -DXDG_STD=TRUE \
      -DDISABLE_SETCAP=ON \
      -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
      -DQT_BUILD=TRUE
    
      # -DPACKAGE_MODE=ON \ ## Cmake claims it doesn't work with Qt builds
  ninja -j$(nproc)
```

<details><summary>Edited AUR PKGBUILD file to build with Qt GUI (this is mostly just bash-like syntax; should be easy to understand):</summary>

```bash
pkgname=pcsx2-git
pkgver=1.7.3341
pkgrel=1
pkgdesc='A Sony PlayStation 2 emulator'
arch=(x86_64)
url=https://www.pcsx2.net
license=(
  GPL2
  GPL3
  LGPL2.1
  LGPL3
)

install=dev9.install

depends=(
  libaio
  libjpeg-turbo
  libpcap
  libgl # For Steam Deck
  xorg-xrandr # For Steam Deck
  libxrender # For Steam Deck
  linux-api-headers # For Steam Deck
  libpulse
  portaudio
  libsamplerate
  sdl2
  soundtouch
  wxgtk3
  wayland
  rapidyaml-git
  qt6-base
)
makedepends=(
  cmake
  git
  xorgproto
  ninja
  swig
  python
  vulkan-headers
  qt6-tools
  doxygen
)
provides=(pcsx2)
conflicts=(pcsx2)
source=(
git+https://github.com/PCSX2/pcsx2.git
git+https://github.com/fmtlib/fmt.git
git+https://github.com/ocornut/imgui.git
git+https://github.com/rtissera/libchdr.git
git+https://github.com/google/googletest.git
git+https://github.com/nih-at/libzip.git
git+https://github.com/facebook/zstd.git
git+https://github.com/mozilla/cubeb.git
git+https://github.com/KhronosGroup/glslang.git
git+https://github.com/libsdl-org/SDL.git
)
sha256sums=(SKIP)
options=('lto')

pkgver()
{
  cd pcsx2
  git describe --tags | sed 's/^v//; s/-dev//; s/-/.r/; s/-g/./'
}

prepare()
{
  cd $srcdir/pcsx2/3rdparty
  git submodule init
  git config submodule.https://github.com/fmtlib/src/fmt.git.url fmt
  git config submodule.https://github.com/rtissera/libchdr.git.url libchdr
  git config submodule.https://github.com/google/googletest.git.url gtest
  git config submodule.https://github.com/mozilla/cubeb.git.url cubeb
  git config submodule.https://github.com/ocornut/imgui.git.url imgui
  git config submodule.https://github.com/KhronosGroup/glslang.git glslang
  git config submodule.https://github.com/libsdl-org/SDL.git SDL
  git config submodule.https://github.com/nih-at/libzip.git libzip
  git config submodule.https://github.com/facebook/zstd.git ztsd
  git submodule update
}

build()
{
  mkdir -p build
  cd build || return
  
  if [ "${CC}" == clang ] || [ "${CXX}" == clang++ ]; then
    export useClang=1
    export clangTidy=1
  fi

  cmake ../pcsx2 \
      -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DUSE_SYSTEM_YAML=TRUE \
      -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
      -DWAYLAND_API=ON \
      -USE_VULKAN=ON \
      -GNinja \
      -DXDG_STD=TRUE \
      -DDISABLE_SETCAP=ON \
      -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
      -DQT_BUILD=TRUE
    
      # -DPACKAGE_MODE=ON \
  ninja -j$(nproc)
}

package()
{
    DESTDIR="${pkgdir}" cmake --install build
}

# vim: ts=2 sw=2 et:
sha256sums=('SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP')
```
</details>

### Expected Behavior

- Compilation should have succeeded without errors

### PCSX2 Revision

v1.7.3341-0-g622914830

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Arch

**Comment by stenzek** [comment]
Sounds like an arch specific issue. We don't have anything to do with the AUR, speak to your packager. I'd also encourage you not to screw with cflags like that, it generally creates more problems than it solves, and is barely beneficial for PCSX2 anyway (majority of CPU time is spent in JITted code).

Our builders use clang for the Qt AppImage and have no such issue, nor do my local clang builds.

**Comment by ThisNekoGuy** [comment]
> I'd also encourage you not to screw with cflags like that, it generally creates more problems than it solves

Arch AUR packages _**always**_ have user CFLAGS without exception 100% of the time without user intervention; to what degree depends on the user but they are _**always**_ there regardless. That said, what you just said would invalidate basically every Arch user since _the Arch repositories are built using the same method as the AUR,_ just in an officially maintained capacity. That kind of response is bizarre to make.

That aside, this is an error related to compiling your source; if the compiler can't find something in the project, just saying "it's Arch's fault" and throwing the issue in the trash without a second thought or any attempt to even remotely identify the issue is pretty ridiculous.

**Comment by Mrlinkwii** [comment]
@ThisNekoGuy  are you using in tree rapidyaml or the package on your pc , i  think this may the the issue  using the rapidyaml package on your pc  , can you try using the in tree rapidyaml

**Comment by ThisNekoGuy** [comment]
I was using the package; I'll try the in-tree

**Comment by stenzek** [comment]
> Arch AUR packages always have user CFLAGS without exception 100% of the time without user intervention; to what degree depends on the user but they are always there regardless.

And you wonder why things don't work. If you want to go against developer advice and rice your CFLAGS for no gain, have fun with things breaking.

> That said, what you just said would invalidate basically every Arch user since the Arch repositories are built using the same method as the AUR, just in an officially maintained capacity.

Except we have nothing to do with the AUR? Like I said above, direct your complaints to them, not us.

**Comment by ThisNekoGuy** [comment]
> And you wonder why things don't work. If you want to go against developer advice and rice your CFLAGS for no gain, have fun with things breaking.

The issue here is git submodules pulling from upstream rapidyaml, not CFLAGs; calm yourself

**Comment by stenzek** [comment]
Well done. You can go fix your build script now. For what it's worth, I'm personally very against submodules for this reason and others, but that's the way PCSX2's setup. I am pushing to slowly transitioning away from it, I've moved imgui in-tree, cubeb will follow soon. But no doubt that'll spark controversy too, because "oh my gosh, you're stopping us from using system packages", even though it's necessary due to local modifications in the library.

I gave you a calm and professional response to your issue, stating that it wasn't an issue on our end, and to speak to the person responsible for packaging it instead. Yet you turned that into an unnecessarily aggressive response. So, thanks for keeping the arrogant arch user meme well and truely alive, please remember that we're all volunteers and as such, the project has limited scope, supporting every configuration under the sun is out of scope - we can only support our own AppImage, and to some extent, source builds following the wiki instructions.

**Comment by ThisNekoGuy** [comment]
To be clear, I was far from rude; that was me being _polite,_ if I wanted to be rude, my approach would have been very different.
Now if you feel your feelings are hurt because I suggested that we cooperate instead of shutting each other down to find a solution, then yes, I'm being "arrogant."

Typically, _real_ Arch arrogance is what you approached me with: telling people to shove off and deal with it themself; I'm fully aware open-source is volunteer-driven and, with that said if you didn't want to participate in this issue, you could have just left it be and let people who actually care about solving community problems handle it instead of arbitrarily deciding things you don't like aren't valid. _**This**_ is what it sounds like to speak professionally, with all due respect.
You needent participate in community non-code problems if you don't like the idea not everyone does things exactly as you do; that mentality is antithetical to open-source.

**Comment by stenzek** [comment]
For the third time, and I'm not sure how I'm being unclear here, **the AUR has nothing to do with us**. I don't know why you even raised the issue here in the first place, because if you follow the instructions on the wiki, it will build perfectly fine.

I don't understand why you expect us to diagnose and fix some script that some packager wrote. We're busy enough with our day jobs, and the little time we have to work on these projects is best spent actually improving the project, rather than fixing other people's stuff.

Besides, the way packaging is *meant* to work, is you report issues to the packager, who determines if it's an upstream issue, and if so, they report it upstream, ideally providing a patch to resolve it. I don't see any of that happening here.

You have the freedom to raise issues, just like we have the freedom to determine they're out of scope and close them. Just because you opened an issue doesn't mean that it will be actioned, if you go in with that expectation, you're going to be disappointed.

**Comment by ThisNekoGuy** [comment]
> Besides, the way packaging is meant to work, is you report issues to the packager, who determines if it's an upstream issue, and if so, they report it upstream, ideally providing a patch to resolve it. I don't see any of that happening here.

What I'm telling you is that *I am the packager;* I've said from the beginning that this was a fork of the original and this issue was to discuss where the problem was coming from and how to resolve it.

Running `git` and `cmake` a couple times shouldn't be as huge a deal as you're making this out to be: it isn't that serious.
That's literally all that's happening here; trying to follow a **distro's standards** to perform that task shouldn't be anywhere near controversial; I could literally pull the equivalent of this topic of conversation from the Arch repos and show you that they look nearly identical and use the exact same build system that you're claiming is the thorn in this project's side for supporting: because God forbid someone try to discuss anything. Yes, the AUR has nothing to do with you, but the odds are that **_someone here is_** this question was for **them**: not you.

**Comment by stenzek** [comment]
I know who the maintainer for the AUR packages is, and they haven't raised any issue with us.

You're blowing things well out of proportion, acting like we're somehow hostile to open source, when in fact, the problem here is very simple. **You are trying to build PCSX2 without a complete source tree.**

Not cloning submodules with PCSX2 will result in the compile failing, which is exactly what happened here. It's like receiving a car from a dealership without any wheels, and then complaining that it doesn't move.

**Comment by ThisNekoGuy** [comment]
> I know who the maintainer for the AUR packages is, and they haven't raised any issue with us.

I haven't significantly changed their process in any way besides switching the the target UI framework with `cmake`; if they haven't raised an issue with you about it, it's likely because they're not even aware of it. But again, that's a GTK3 package; approaching them about this modification would likely prompt a "I'm not obligated to maintain a second package" response, thus why I am **_here_** asking a _community_ instead of an _individual_ that may or may not ever answer the question. Put more eyes on it, and it will be seen.

> You are trying to build PCSX2 without a complete source tree.

Evidently, we know this now, but only after you berated me for trying to learn how to resolve this.

**Comment by stenzek** [comment]
> I haven't significantly changed their process in any way besides switching the the target UI framework with cmake; if they haven't raised an issue with you about it, it's likely because they're not even aware of it.

StringUtil is used for both wx and qt. I know they also have a Qt package, because we respectfully asked them to stop distributing it before it was ready for production usage, as well as before we had our own AppImage builders in place.

> approaching them about this modification would likely prompt a "I'm not obligated to maintain a second package" response

So, you don't want to bother them, but you're happy to bother us? (see below)

> thus why I am here asking a community instead of an individual that may or may not ever answer the question

Issue trackers are for reporting issues to developers and maintainers. If you want to pose questions to the community, I'd suggest using the Discussions section, the Discord server, or the PCSX2 forums.

**Comment by ThisNekoGuy** [comment]
> StringUtil is used for both wx and qt.

I didn't dispute that; _I said: that I only touched the cmake arguements to use the new UI_

> Issue trackers are for reporting issues to developers and maintainers. If you want to pose questions to the community, I'd suggest using the Discussions section, the Discord server, or the PCSX2 forums.

See, **_that_** is a far more productive suggestion than everything you've said thus far instead of telling someone to shove-off from the start just to make a clear distinction on what's considered appropriate for the issue template and what **named** alternatives there are instead of initially suggesting I do the same to the AUR maintainer that you feel I just did to you.

**Comment by stenzek** [comment]
Great. We're done here then.
